File: CODIGO 119/8.scala
object FibonacciBrainfuck {
  def generateBrainfuckCode(N: Int, f0: Int, f1: Int): String = {
    val bfCode = new StringBuilder()

    // Inicializar celdas de memoria para f0 y f1
    bfCode.append("[-]") // Inicializar la celda de memoria en 0
    bfCode.append("+" * f0)
    bfCode.append(">" * f1)

    // Generar secuencia Fibonacci
    bfCode.append("[")
    bfCode.append("->+[-<+>]")
    bfCode.append(">" * f1)
    bfCode.append("<<<<<-]")
    bfCode.append("[" + "+" * f1 + ">" * f0 + "<-]>[->+<][>+<-]<[<+>>+<-]>[->+<]")

    // Imprimir N valores de la secuencia
    bfCode.append(">" * (N - 1))
    bfCode.append("[")
    bfCode.append(">++++++[<++++++++>-]<.[-]<[-]<")
    bfCode.append("<" * (N - 1))
    bfCode.append("-]")

    bfCode.toString()
  }

  def main(args: Array[String]): Unit = {
    val N = 5
    val f0 = 1
    val f1 = 3

    val bfCode = generateBrainfuckCode(N, f0, f1)
    println(bfCode)
  }
}

File: CODIGO 119/8.cpp
#include <iostream>
#include <string>

std::string generateBrainfuckCode(int N, int f0, int f1) {
    std::string bfCode;

    // Inicializar celdas de memoria para f0 y f1
    bfCode += "[-]"; // Inicializar la celda de memoria en 0
    bfCode += std::string("+", f0);
    bfCode += std::string(">", f1);

    // Generar secuencia Fibonacci
    bfCode += "[";
    bfCode += "->+[-<+]";
    bfCode += std::string(">", f1);
    bfCode += "<<<<<-]";
    bfCode += "[" + std::string("+", f1) + std::string(">", f0) + "<-]>[->+<][>+<-]<[<+>>+<-]>[->+<]";

    // Imprimir N valores de la secuencia
    bfCode += std::string(">", N - 1);
    bfCode += "[";
    bfCode += ">++++++[<++++++++>-]<.[-]<[-]<";
    bfCode += std::string("<", N - 1);
    bfCode += "-]";

    return bfCode;
}

int main() {
    int N = 5;
    int f0 = 1;
    int f1 = 3;

    std::string bfCode = generateBrainfuckCode(N, f0, f1);
    std::cout << bfCode << std::endl;

    return 0;
}
